### PR TITLE
Modpoller

### PR DESCRIPTION
**Pull request by IgnatovFedor**

### CODE REVIEW

File: utils/stand_monitoring/poller/config.json [hunk 1]
```diff
@@ -1,69 +1,29 @@
...
-  },
-  "response_patterns": {
-    "default": {
-      "code": 200
-    },
-    "https://2242.lnsigo.mipt.ru/botConvAIrouterprobing/getUpdates": {
-      "code": 401
-    }
+  "stands": {
+    "https://7003.lnsigo.mipt.ru/answer": "stand_paraphraser_ru",
```
Comment (litinsky) [inline utils/stand_monitoring/poller/config.json]: Model name should be key, URL - value.

File: utils/stand_monitoring/poller/config.json [hunk 2]
```diff
@@ -1,69 +1,29 @@
 {
   "general": {
     "polling_interval": 120,
-    "notification": "slack"
+    "request_timeout": 10,
+    "notification": "slack",
+    "response_pattern": 200
```
Comment (litinsky) [inline utils/stand_monitoring/poller/config.json]: There is no need in parameterizing respose code, its stable for all http protocol and can be hardcoded.

File: utils/stand_monitoring/poller/poller.py [hunk 3]
```diff
@@ -1,88 +1,76 @@
...
-from requests import Response
 
 
 config_path = './config.json'
 with open(config_path, 'r') as f_config:
     config = json.load(f_config)
 
 
-def probe(urls: dict, payload: dict, response_patterns: dict):
+def probe(stands: dict, request_timeout: float, response_pattern: int):
```
Comment (litinsky) [inline utils/stand_monitoring/poller/poller.py]: 1) Bad naming. We have one stand and several services. `services`, for examle, is better;
2) No need in response pattern, see above.

File: utils/stand_monitoring/poller/poller.py [hunk 4]
```diff
@@ -1,88 +1,76 @@
...
-            try:
-                response = requests.post(url, json=payload[payload_type])
-                probe_result[url] = assert_response(response, pattern)
-            except Exception:
-                probe_result[url] = False
-
-    for url in urls_get:
-        pattern = response_patterns[url] if url in response_patterns.keys() else response_patterns['default']
+
+    for url, model in stands.items():
```
Comment (litinsky) [inline utils/stand_monitoring/poller/poller.py]: model_name - key, url - value, see above...

File: utils/stand_monitoring/poller/poller.py [hunk 5]
```diff
@@ -1,88 +1,76 @@
...
+        stand_name = ' '.join([model, url])
         try:
-            response = requests.get(url)
-            probe_result[url] = assert_response(response, pattern)
+            response = requests.post(url+'/poller', json={}, timeout=request_timeout)
```
Comment (litinsky) [inline utils/stand_monitoring/poller/poller.py]: `url+'/poller'` - is akward, using f-strings is better style.

File: utils/stand_monitoring/poller/poller.py [hunk 6]
```diff
@@ -1,88 +1,76 @@
...
 
 
-def notify(bad_urls: list):
+def notify(stands: dict, first_notification=False):
     channel = config['general']['notification']
     channel_config = config['notification'][channel]
 
-    bad_urls_str = '\n'.join(bad_urls)
-    notification = f'Following stand endpoints are unreachable:\n\n{bad_urls_str}'
+    good_stands = '\n'.join([stand for stand, status in stands.items() if status])
```
Comment (litinsky) [inline utils/stand_monitoring/poller/poller.py]: Naming: `good_stands` -> `up_stands`, `bad_services` -> `down_services`

File: utils/stand_monitoring/poller/poller.py [hunk 7]
```diff
@@ -1,88 +1,76 @@
...
+    bad_stands = '\n'.join([stand for stand, status in stands.items() if not status])
+    notification = []
+    if first_notification:
```
Comment (litinsky) [inline utils/stand_monitoring/poller/poller.py]: These 3 messages are better to be parametrized in config file.

File: utils/stand_monitoring/poller/poller.py [hunk 8]
```diff
@@ -1,88 +1,76 @@
...
+        notification.append('THIS IS FIRST NOTIFICATION AFTER LAUNCH.')
```
Comment (litinsky) [inline utils/stand_monitoring/poller/poller.py]: Wording: `THIS IS FIRST NOTIFICATION AFTER LAUNCH` -> `DEMO STAND SERVICES HEALTH CHECK SCRIPT WAS LAUNCHED`

File: utils/stand_monitoring/poller/poller.py (hunk 7 above)
Comment (IgnatovFedor) [reply utils/stand_monitoring/poller/poller.py]: I moved three messages (launch, before up services and down) to config file.

**Pull request merged**